packages/dashboard/src/managers/task-manager.ts

```diff
@@ -0,0 +1,53 @@
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import EventEmitter from 'eventemitter3';
+
+type Events = {
+  updated: [];
+};
+
+export default class TaskManager extends EventEmitter<Events> {
+  tasks(): RomiCore.TaskSummary[] {
+    return Array.from(Object.values(this._tasks));
+  }
+
+  startSubscription(transport: RomiCore.Transport) {
+    this._subscriptions.push(
+      transport.subscribe(RomiCore.taskSummaries, (taskSummary) => {
+        this._tasks[taskSummary.task_id] = taskSummary;
+        this.emit('updated');
+      }),
+    );
+  }
+
+  stopAllSubscriptions(): void {
+    this._subscriptions.forEach((sub) => sub.unsubscribe());
+  }
+
+  static getActorFromStatus(status: string) {
```

Comment:
This looks like a hack, can you add a `TODO` and a issue (here and at `rmf_core`) to provide a reliable way to retrieve the actor from the message?

Reply:
Sure, sounds good! I'll create and issue and add a todo

Reply:
https://github.com/osrf/rmf_core/issues/205